### ISSUE TITLE
ERROR (EidosScript::Match): unexpected token '#0.'

### ISSUE THREAD

**Issue by Apdulai** [grahamgower/genomatnn#58]
Hi @grahamgower 

I have another issue encountered while trying to run the tutorial on human data. I got an error after running this line: `$ genomatnn sim -n 1000 Nea_to_CEU.toml`. Here is the error message:

Thanks for help !

![image](https://user-images.githubusercontent.com/69516027/158234123-7e889277-ed07-440c-91e4-744c1a280540.png)

**Comment by grahamgower** [comment]
Hi @Apdulai. I went through the instructions from scratch and couldn't reproduce this problem. I reckon that if you run the tests again, they will fail. From your error message, I can see that your installation is using a version of stdpopsim that is installed in `~/.local/lib/python3.8/site-packages/stdpopsim/`. This is very likely to be the cause of your problems, because genomatnn needs a very specific version of stdpopsim -- if its too old then it won't include the selection framework, and if its too new then there are changes which haven't yet been accounted for in genomatnn. For this reason, when you install genomatnn with `python setup.py genomatnn`, it asks for a specific stdpopsim version (the "selection" branch from my stdpopsim github fork, which is known to work).

Furthermore, Python packaging is a nightmare... When you install `<foo>` with `pip install <foo>`, it's not obvious where pip will install `<foo>`. If you type `pip -V` that might give an indication (although I'm not sure if it's always correct). If you install things with `sudo pip install ...` then the package is install systrem wide, probably somewhere like /usr/lib/python3.8/site-packages. If you don't use sudo, then pip will try to install under your $HOME, likely in `$HOME/.local/lib/python3.8/...`. However, if you have a conda environment activated, then `pip install` will install into the conda environment. Unfortunately, if you have things installed in both `~/.local/lib/...` and in your conda environment, there can be conflicts between the two (as you have observed here).

To fix this, you need to uninstall stdpopsim from `~/.local/lib`. One way to do this is to deactivate your conda environment, then run `pip uninstall stdpopsim` repeatedly until you recieve an error that stdpopsim isn't installed (it is possible that a package can be installed multiple times in conflicting ways, so repeatedly uninstalling *may* be necessary!). But then, its possible that you have other things installed in `~/.local/lib/python3.8/...` that conflict. So really, the most robust solution is to (re)move your `~/.local/lib/python3.8` folder, and create a fresh genomatnn conda environment following the instructions in the README.

Sorry, Python packaging really is bad for both users and developers. :(

**Comment by Apdulai** [comment]
Indeed, you were right. Running the `pip uninstall stdpopsim` command, it appears that it was not installed in the base system, but just in my home. Moreover, there were also two folders of stdpopsim in `$HOME/.local/lib/python3.8/site-packages`.  Probably in conflict!


![image](https://user-images.githubusercontent.com/69516027/158419792-54497721-8d26-4c3a-a3bb-3ddb746073a0.png)

I don't have `sudo` rights, I am performing the analysis on an HPC. 

Finally, I had to delete the folder `$HOME/.local/lib/python3.8` and the previously created environment, and start the installation step by step from the README. I did a test run and all the steps were done successfully until the prediction of the adaptive introgression.

Thanks a lot for your support

Sincerely,
